src/System.Windows.Forms/src/System/Windows/Forms/TaskDialog.cs

```diff
@@ -812,10 +812,10 @@ internal void UpdateCaption(string? caption)
             {
                 caption = Path.GetFileName(
                     UnsafeNativeMethods.GetModuleFileNameLongPath(NativeMethods.NullHandleRef)
-                    .ToString());
+                                       .ToString());
             }
 
-            User32.SetWindowTextW(Handle, caption);
+            User32.SetWindowTextW(Handle, caption ?? string.Empty);
```

Comment:
@kpreisser can you think of any issues with this? `Path.GetFileName() => string?`, so `caption` may be `null`. I could also annotate if with `caption!` but it didn't feel right...

Reply:
Actually, `caption` should not be `null` here, because `Path.GetFileName()` only returns `null` if its argument was `null`.

The issue is that `TaskDialogPage.IsNativeStringNullOrEmpty` is missing an `[NotNullWhen]` annotation for its parameter to indicate that the string is not `null` when it returns `false`. It should be changed as follows:
```c#
internal static bool IsNativeStringNullOrEmpty([NotNullWhen(false)] string? str)
```

With that change, the code in `TaskDialog.UpdateCaption` doesn't need to be changed:
```suggestion
            User32.SetWindowTextW(Handle, caption);
```

Reply:
That worked! Thank you